### PR TITLE
add shigeifinder 1.3.5; implement builder stage to reduce image size

### PR DESCRIPTION
**Pull request by kapsakcj**
This PR adds shigeifinder 1.3.5, which adds the ability to print the version with `shigeifinder --version` and has a small bug fix for tmp directory introduced in 1.3.4 (which is why I skipped over 1.3.4).

I have significantly changed the structure of the dockerfile by adding a `builder` stage to the dockerfile to remove all the stuff needed for compiling samtools & bwa C code. Took inspiration from @Kincekara 's recent PR for updating samtools! TY Kutluhan!

The updated multi-stage build has significantly reduced the size of the final `app` stage docker image. Uncompressed image size is ~592MB smaller than the previous docker image `staphb/shigeifinder:1.3.3` 🎉 
```bash
$ docker images | grep shigeifinder
shigeifinder         1.3.5-test       1cd5b4b813fb   2 hours ago     893MB
shigeifinder         1.3.5-app        792f4821434e   2 hours ago     297MB
shigeifinder         1.3.5-builder    da771942919d   2 hours ago     730MB
staphb/shigeifinder  1.3.3            876c3731b3d8   5 months ago    889MB
```

Changes from previous dockerfile:

- added `builder` stage for compiling `samtools` and `bwa` code
- updated `app` stage to remove all dependencies for compiling C code (`gcc`, `zlib1g-dev`, etc.)
  - `bwa` and all `samtools` executables now located in `/usr/local/bin` as this is globally accessible and on the $PATH for all users.
- updated `test` stage to test the new `shigeifinder --version` flag.

⚠️  NOTE: upgrading to `ubuntu:jammy` will break the `bwa` C code compiliation process as the `gcc` version installed via apt for ubuntu:jammy is newer and has changed how variable declarations are handled. We will likely need to use `ubuntu:focal` or an older release for any dockerfile that installs bwa 0.7.17 from source. If they ever release another version of `bwa`, then we should be fine to upgrade, the fix just is not in v0.7.17

Pull Request (PR) checklist:
- [x] Include a description of what is in this pull request in this message.
- [x] The dockerfile successfully builds to a test target for the user creating the PR. (i.e. `docker build --tag samtools:1.15test --target test docker-builds/samtools/1.15` )
- [x] Directory structure as name of the tool in lower case with special characters removed with a subdirectory of the version number (i.e. `spades/3.12.0/Dockerfile`)
   - [x] (optional) All test files are located in same directory as the Dockerfile (i.e. `shigatyper/2.0.1/test.sh`)
- [x] Create a simple container-specific [README.md](https://github.com/StaPH-B/docker-builds/blob/master/.github/workflow-templates/readme-template.md) in the same directory as the Dockerfile (i.e. `spades/3.12.0/README.md`)
   - [x] If this README is longer than 30 lines, there is an explanation as to why more detail was needed
- [x] Dockerfile includes the recommended [LABELS](https://github.com/StaPH-B/docker-builds/blob/master/dockerfile-template/Dockerfile#L8-L18) 
- [x] Main [README.md](https://github.com/StaPH-B/docker-builds/blob/master/README.md) has been updated to include the tool and/or version of the dockerfile(s) in this PR
- [x] [Program_Licenses.md](https://github.com/StaPH-B/docker-builds/blob/master/Program_Licenses.md) contains the tool(s) used in this PR and has been updated for any missing

### CODE REVIEW

File: shigeifinder/1.3.5/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+LABEL base.image="ubuntu:focal"
+LABEL dockerfile.version="1"
+LABEL software="ShigEiFinder"
+LABEL software.version="${SHIGEIFINDER_VER}"
+LABEL description="Cluster informed Shigella and EIEC serotyping tool from Illumina reads and assemblies"
+LABEL website="https://github.com/LanLab/ShigEiFinder"
+LABEL license="https://github.com/LanLab/ShigEiFinder/blob/main/LICENSE"
+LABEL maintainer="Curtis Kapsak"
+LABEL maintainer.email="curtis.kapsak@theiagen.com"
```
Comment (erinyoung) [inline shigeifinder/1.3.5/Dockerfile:22]: These labels are not making it to the test stage of the build. I don't have a lot of time to experiment with this, but I think you want to put these in `app` instead.

Comment (kapsakcj) [reply shigeifinder/1.3.5/Dockerfile:22]: got it thank you. I will make the adjustment shortly

**Comment by kapsakcj** [comment]
OK issues with the LABELs has been resolved. Good to know that they need to be within the `app` stage and not in any `builder` or other stage not included within `app`.

**Comment by Kincekara** [comment]
Nice touch :clap:

**Comment by kapsakcj** [comment]
Docker image is available on my personal dockerhub account for testing: `kapsakcj/shigeifinder:1.3.5`

https://hub.docker.com/r/kapsakcj/shigeifinder/tags

**Comment by erinyoung** [comment]
Are you waiting for Kevin to review this, or can I?

**Review by erinyoung — APPROVED** [review]
Thank you for this PR!

I have no additional suggestions. Your tests appeared to work, and I think it's great that the container is so much smaller.

I am going to 
1. merge this PR
2. push to dockerhub and quay with tag '1.3.5'
3. overwrite the 'latest' tag

**Comment by erinyoung** [comment]
You can watch the status on the deployment here : https://github.com/StaPH-B/docker-builds/actions/runs/5446967381

I'll let you delete your branch if you're done with it.

**Pull request merged**